### ISSUE TITLE
Failed run is not shown in dashboard (The job failed with Client error '404 Not Found')

### ISSUE THREAD

**Issue by yaronlevi** [PrefectHQ/prefect#10399]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

In the main dashboard, a single failed job has happened (you can the number "1" in the label, but the run itself is not shown:

![CleanShot 2023-08-14 at 09 46 10@2x](https://github.com/PrefectHQ/prefect/assets/5398977/a603018c-0cfd-437a-8dd5-013ac420e343)

If it helps, [this](https://app.prefect.cloud/account/8eed9803-456a-4126-a7f7-074aa44aa1b2/workspace/8ff919f1-a2c3-4660-9ab5-66b57758d46d/flow-runs/flow-run/2be14b57-573f-40a1-b949-e864c64f6095) is the link to the failed job.

And [this](https://app.prefect.cloud/account/8eed9803-456a-4126-a7f7-074aa44aa1b2/workspace/8ff919f1-a2c3-4660-9ab5-66b57758d46d/dashboard?flow-run-state=failed) is the link to the dashboard.

Also this is the error inside that failed job:

![CleanShot 2023-08-14 at 09 53 36@2x](https://github.com/PrefectHQ/prefect/assets/5398977/2ecc5f0f-1846-4133-b3b9-c02e2462b2a8)

Error message:
```
Submission failed. PrefectHTTPStatusError: Client error '404 Not Found' for url 'https://api.prefect.cloud/api/accounts/8eed9803-456a-4126-a7f7-074aa44aa1b2/workspaces/8ff919f1-a2c3-4660-9ab5-66b57758d46d/flows/39069092-7ea2-4887-b647-07b955f63841' Response: {'detail': 'Flow not found'} For more information check: https://httpstatuses.com/404
```

Again, I don't know if the 404 error and the fact that this failed job is not shown in the dashboard are related.
Maybe it's two different issues.

### Reproduction

```python3
Happens in Prefect Cloud.
```

### Versions

```Text
Prefect Cloud
```

**Comment by pleek91** [comment]
@yaronlevi any js errors in your browser's devtools?

**Comment by yaronlevi** [comment]
@pleek91 No. Nothing special or related in the devtools logs.

**Comment by desertaxle** [comment]
It looks like the record of the flow may have been deleted during execution. @yaronlevi are you able to see the flow for this flow run in the UI?

**Comment by yaronlevi** [comment]
@desertaxle Yeah, the record was probably deleted... getting a 404 when pressing the link.
